components/esp_eth/src/esp_eth_phy_ksz8041.c

```diff
@@ -0,0 +1,354 @@
+// Copyright 2019 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <string.h>
+#include <stdlib.h>
+#include <sys/cdefs.h>
+#include "esp_log.h"
+#include "esp_eth.h"
+#include "eth_phy_regs_struct.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "driver/gpio.h"
+
+static const char *TAG = "ksz8041";
+#define PHY_CHECK(a, str, goto_tag, ...)                                          \
+    do                                                                            \
+    {                                                                             \
+        if (!(a))                                                                 \
+        {                                                                         \
+            ESP_LOGE(TAG, "%s(%d): " str, __FUNCTION__, __LINE__, ##__VA_ARGS__); \
+            goto goto_tag;                                                        \
+        }                                                                         \
+    } while (0)
+
+/***************Vendor Specific Register***************/
+/**
+ * @brief LMDR(LinkMD control/status Register)
+ *
+ */
+typedef union {
+    struct {
+        uint32_t cd_test_enable : 1;   /* Cable Diagnostic Test Enable */
+        uint32_t cd_test_result : 2;   /* Cable Diagnostic Test Result */
+        uint32_t reserved : 4;         /* Reserved */
+        uint32_t cable_fault_cnt : 9;  /* Cable Fault Counter, 1u = 0.4m */
+    };
+    uint32_t val;
+} lmdr_reg_t;
+#define ETH_PHY_LMDR_REG_ADDR (0x1D)
```

Comment:
Looks like this register is used in test and diagnostic cases, but our driver currently doesn't support this feature. So suggest to remove it for now.

Reply:
Yes. Those regs are mainly for line diagnostic, somehow useful. Just removed, for now.